client/src/main/java/com/paypal/selion/platform/grid/SeleniumGridListener.java

```diff
@@ -125,6 +126,8 @@ public void beforeInvocation(IInvokedMethod method, ITestResult testResult) {
             }
         } catch (Exception e) { // NOSONAR
             logger.log(Level.WARNING, "An error occurred while processing beforeInvocation: " + e.getMessage(), e);
+            //Failing the test case, because pre validation/configuration got failed
+            Assert.fail(e.getMessage(), e);
```

Comment:
@sundaramrajendran 

IMO, `Asserts` are to be used ONLY in tests and Not within a framework.